packages/wallet/test/services/TransactionsTracker.test.ts

```diff
@@ -280,6 +280,42 @@ describe('TransactionsTracker', () => {
         pagination: { limit: 25, startAt: 0 }
       });
     });
+
+    describe('distinct transaction sets in latest stored block vs new blocks', () => {
```

Comment:
@mkazlauskas , @AngelCastilloB , please review these proposed scenarios to cover in unit tests.
If they make sense and improve coverage I will implement them.

Reply:
Looks good, just don't quite understand when/how this could happen `reversed order transactions plus new tx are re-emitted, but not considered rollbacks`

Reply:
It is far fetched, but I was thinking of a scenario where a competing SP has txId 3 in the mempool, then gets a mempool update with txId 2, then another with txId 1, and creates a fork with the 3 transactions <3 2 1>.
I'm not sure if that is even possible tbh 😅

Reply:
Looks great @mirceahasegan , I think we just need to add the case where there are no changes:

```
// latestStoredBlock  <1 2 3>
// fetched Txs            <1 2 3>
```

Or is it that this is already covered by the already existing tests?

Reply:
Thanksk @mirceahasegan , better safe than sorry :sweat_smile:

Reply:
I would say it's already covered in the other tests, for single transaction blocks.
I'll add one for the multi-transaction block scenario too, just to be safe.

Reply:
Unit tests are implemented in ef247eaf2c51547162ec4f997ec4d34b600970c0.
The commit also contains a fix for an issue discovered in tx tracker.
There still are ~3 tests that are failing and we need to investigate if they're bugs or test issues after updating implementation